docs/source/iam.md

```diff
@@ -0,0 +1,23 @@
+# Using Scaleway IAM
+
+To deploy the gateway using Scaleway IAM, we recommend the following setup:
+
+1. Create a new Scaleway project
+2. Create an IAM application scoped to this project
+3. Add a policy with this application as principal
+4. Add the following rules to the policy:
+    - Serverless -> `ContainersFullAccess`
+    - Managed services -> `ObservabilityFullAccess`
+    - Managed databases -> `RelationalDatabasesFullAccess`
+    - Security and identity -> `SecretManagerFullAccess`
+5. Create an API key for the application
```

Comment:
I would also love a small note about Scaleway Projects. Because technically, the least privileged approach would be to deploy the gateway on its own project by creating a new project and scoping the IAM permissions on this project only. As an example, there's no risk for the token with `RdbFullAcess` to delete the databases contained in the default project.
Like this: 
![image](https://github.com/scaleway/serverless-gateway/assets/49867608/9e440fb4-22be-4ab4-9b18-e99274a90123)